harness/determined/core/__init__.py

```diff
@@ -25,4 +25,20 @@
     _PreemptionWatcher,
     PreemptMode,
 )
-from determined.core._context import Context, init, _dummy_init
+from determined.core._heartbeat import (
+    Heartbeat,
+    ManagedTrialHeartbeat,
+    UnmanagedTrialHeartbeat,
```

Comment:
Are the Heartbeat symbols user-facing?  Would they ever either instantiate one or call any of its methods?

Reply:
Okay I think I finally understand that the rule is: every class or function which will not be used by an end-user, must start with an underscore.
we are *not* using the "internal" notion from the style guides which is `“Internal” means internal to a module, or protected or private within a class.`

Reply:
Yes, that is correct.

@wes-turner and I actually talked about this at length.  My stance is: python doesn't have the right semantics to distinguish what is "private to this module" vs "private to determined developers" vs "public to end users", and so I (strongly) prefer that we use underscore to mean "not public to end users", since I think of three categories, that's by far the most important category to distinguish.